doc/design/dynamic_rnn.md

```diff
@@ -0,0 +1,126 @@
+# Implementation Doc: Dynamic RNN
+
+## A glance of Dynamic RNN
+
+A common neural network structure called recurrent neural network(`RNN` for short), which there is a directed circle in the neural network model. RNN can use a internal memory to process arbitrary sequences of inputs.
+
+PaddlePaddle Fluid directly represents the `directed circle` in the `ProgramDesc`, since we do not use directed acyclic graph to represent our model. The `ProgramDesc` just like the AST of a programming language, which describes the computation instructions for training a neural network. We use arrays and a while loop to describe the training process of an RNN. The C++ code below demonstrates the forward logic of RNN which PaddlePaddle Fluid generates in `ProgramDesc`.
```

Comment:
training process --> training/inference process 
The C++ code below demonstrates the forward logic of RNN which PaddlePaddle Fluid generates in `ProgramDesc` --> The C++ code below demonstrates the forward logic of RNN generated in `ProgramDesc` of PaddlePaddle Fluid.

Reply:
Done